### ISSUE TITLE
cannot replicate NELL-995 MINERVA result

### ISSUE THREAD

**Issue by gstoica27** [salesforce/MultiHopKG#9]
Hi, 

Going through the code, it seems that 'point' is equivalent to MINERVA. While I can achieve expected performance on the remaining datasets, I am unable to get anywhere near reported performance on NELL-995. I have tried using the exact config provided in conjunction with the --test process, and with the exact parameters given by MINERVA. 

Below is my performance after training with the default config:
Dev:
Hits@1 = .727
Hits@3 = .893
Hits@5 = .895
Hits@10 = .901
MRR = .812
Test:
Hits@1 = .500
Hits@3 = .694
Hits@5 = .742
Hits@10 = .782
MRR = .611
And with exact hyperparameters (changing beta to .06 and gamma to 0):
Dev:
Hits@1 = .479
Hits@3 = .750
Hits@5 = .847
Hits@10 = .873
MRR = .63
Test:
Hits@1 = .380
Hits@3 = .585
Hits@5 = .676
Hits@10 = .775
MRR = .498

Is there something I am missing when it comes to training the model?

**Comment by todpole3** [comment]
Sorry for missing this issue. I was able to reproduce the NELL-995 results following our instructions so I'm happy to help.

First of all, did you train the NELL-995 ConvE model before training the RL model?

**Comment by todpole3** [comment]
Closing this issue due to lack of activity. Please reopen if you continue to experience this problem.

**Comment by David-Lee-1990** [comment]
Hi，i checked the source data "raw.kb" and "test.triples" of  NELL-995. I find that 26.9% of the triples's head entity in test data is missing in the raw.kb data. So how can hit@K  be larger that 73.1%?

**Comment by todpole3** [comment]
@David-Lee-1990 I noticed that you closed issue #10 I'm bewildered by the stats you reported too (if they were correct). What was the findings that lead you to solve the problem?

**Comment by David-Lee-1990** [comment]
@todpole3  Just now, I  go through the detail of your code. I find you just delete those triples with unseen head or tail entity in the test data.  So in this way, the result would be correct.

**Comment by todpole3** [comment]
Good catch. We adopted the practice from previous work (https://arxiv.org/pdf/1711.05851.pdf ).

**Comment by David-Lee-1990** [comment]
@todpole3  By the way, I find the hit@K used in your paper is filtered,i.e. removing the other correct answers from the predicted answers. However, MINERVA‘s hit@K results are not filtered ( got this by reading their code).

**Comment by todpole3** [comment]
@David-Lee-1990 Hmm, I believe both work filters when computing hits@k (otherwise the numbers does not make much sense). Would you mind posting the evaluation code segment you read?

**Comment by David-Lee-1990** [comment]
sorry,  my mistake,  MINERVA filtered by replace the other correct answers by PAD when return next actions.

**Comment by gstoica27** [comment]
Hi, apologies for never following up on the original comment. 
I have recently been trying again to replicate the results reported with the implementation of MINERVA by the 'point' method but am still having trouble obtaining near reported performance. 

I've tried using the 'point' method from NELL-995.sh right out of the box as well as changing its parameters to exactly mirror MINERVA's hyperparameter setup but haven't been able to get close to their reported performances. For instance, the best performance came from NELL-995 right out of the box with (in decimal form):

HITS@1: .544
HITS@3: .735
HITS@5: .775
HITS@10: .811
MRR: .642

To answer the question from a few months ago regarding training ConvE first, I did not, but my understanding is that ConvE/embedding based methods would be needed to train the reward shaping version of MultiHopKG, instead of the 'point' version. Is this correct? Or should I still train ConvE before training 'point' for NELL-995? 

Thanks very much for all the help!
